services/service-admin/src/services/feature.service.ts

```diff
@@ -87,4 +89,24 @@ export class FeatureService {
       });
     });
   }
+
+  async createCheckoutSession(unitAmountDecimal: string): Promise<string | Error> {
+    const stripe = new Stripe(config.STRIPE_SECRET_KEY);
+    const session = await stripe.checkout.sessions.create({
+      line_items: [{
+        price_data: {
+          currency: 'nzd',
+          product_data: {
+            name: 'One-time Donation',
+          },
+          unit_amount_decimal: unitAmountDecimal,
+        },
+        quantity: 1,
+      }],
+      mode: 'payment',
+      ui_mode: 'embedded',
+      return_url: `${config.SPKEASY_HOST}/donate/thanks`,
+    });
+    return session.client_secret ?? new Error("Stripe API call failed");
```

Comment:
```suggestion
return session.client_secret ?? new Error("Stripe API call did not return a client secret");
```

I'm _fairly_ certain that our error middleware will turn an unknown error into an `Internal Server Error` so we're not giving away details (and if it doesn't, it should).

But by giving a little more detail here, it'll be a little easier for whoeever ends up debugging the error message that appears in the logs